### ISSUE TITLE
IPWB-Compatible collection archiving task proof-of-concept

### ISSUE THREAD

**Issue by b5** [archivers-space/task-mgmt#5]
Connecting @machawk1 & @oduwsdl: https://github.com/oduwsdl/ipwb/issues/211

We should define a task that:
1. Start with a user-generated collection of URLs. Allow users to fire off a "task" that will...
2. Generate a WARC of that collection using https://github.com/datatogether/warc
3. Generate an IPWB-Compatible CDXJ file using https://github.com/datatogether/cdxj
4. Put all of that on IPFS
5. Demo the WARC in IPWB.

**Comment by machawk1** [comment]
When generating WARCs using user-agents other than browsers, it's possible that the capture may not be comprehensive to the extent needed for accurate replay. For example, if I `wget -p -k --warc-file=myarchive uriWithLotsOfJS.com`, wget may not grab the representations of resources that are conventionally surfaced via JS. This could also be applied to dynamically built URIs, URIs within resources with URIs within, etc.

For the sake of a demo, it might be useful to first examine the potential for missed URIs when dereferencing them while creating the WARCs (the lib might need to handle this).

**Comment by b5** [comment]
Interesting. Would you recommend applying user-agent spoofing at all? I'm thinking of [this approach](https://github.com/oduwsdl/memgator/blob/master/main.go#L162).

Either way, noted! Part of me thinks we should build / seek out some sort of "archiving obstacle course" to run tests against. If this doesn't already exist, it seems like it'd be worth having around for a number of different projects

**Comment by machawk1** [comment]
@b5 It's not necessarily the user-agent string but the capability of the agent. If the agent does not execute JS, some resource representations may not be surfaced and thus not archived by the tool.

Awhile back I put together the [Archival Acid Test](http://acid.matkelly.com/) (more info in the [short paper](http://www.cs.odu.edu/~mkelly/papers/2014_dl_acid.pdf)) to evaluate existing crawlers/archival tools but that was a few years ago. Since then, I know the [UK Web Archive](https://github.com/ukwa/wren) started writing some evaluation mechanism and I believe @N0taN3rd is in the process of rewriting and extending my previous tests.

**Comment by N0taN3rd** [comment]
@machawk1 @b5 Yes I  am currently compiling a `Good Luck Youll Need It` list with implementation 
But until that is finished you can have some fun with [iframe madness](http://www.cs.odu.edu/~jberlin/funtimes2/) and a mini [replay test](https://n0tan3rd.github.io/replay_test/) for [2017-03-09: A State Of Replay or Location, Location, Location](http://ws-dl.blogspot.com/2017/03/2017-03-09-state-of-replay-or-location.html)

iframe madness is currently unarchivable ([Internet Archive](http://web.archive.org/web/20170519051421/http://www.cs.odu.edu/~jberlin/funtimes2/)) for all non high-fidelity archives

IPWB is high-fidelity :+1:
